pkg/auth/docker/client.go

```diff
@@ -64,6 +64,39 @@ func NewClient(configPaths ...string) (auth.Client, error) {
 	}, nil
 }
 
+// NewClientWithDockerFallback creates a new auth client
+// which falls back on Docker's default config path.
+// This allows support for ~/.docker/config.json as a fallback,
+// as well as support for the DOCKER_CONFIG environment variable.
+func NewClientWithDockerFallback(configPaths ...string) (auth.Client, error) {
+	if len(configPaths) == 0 {
+		return nil, errors.New("must provide at least one config path")
```

Comment:
Would we want to call `NewClient` when no config paths have been provided? Or is this a way to enforce the end user call `NewClient` if they do not provide any `configPaths`

Reply:
Yes, done